ccm_web/client/src/components/ConfirmationTable.tsx

```diff
@@ -0,0 +1,186 @@
+import React, { useEffect, useState } from 'react'
+import { IconButton, makeStyles, Paper, Table, TableBody, TableCell, TableContainer, TableFooter, TableHead, TablePagination, TableRow, useTheme } from '@material-ui/core'
+import { KeyboardArrowRight, KeyboardArrowLeft, LastPage as LastPageIcon, FirstPage as FirstPageIcon } from '@material-ui/icons'
+import PropTypes from 'prop-types'
+
+const useStyles = makeStyles((theme) => ({
+  root: {
+    flexShrink: 0,
+    marginLeft: theme.spacing(2.5)
+  }
+}))
+
+interface StudentGrade {
+  rowNumber: number
+  uniqname: string
+  grade: string
+
+}
+
+interface ConfirmationTableProps {
+  grades: StudentGrade[]
+}
+
+interface TableHeaderColumnInfoShouldUseMatUIType {
+  id: keyof StudentGrade
+  label: string
+  minWidth: number
+  align?: 'left' | 'right' | undefined
+}
+
+const columns: TableHeaderColumnInfoShouldUseMatUIType[] = [
+  { id: 'rowNumber', label: 'Row Number', minWidth: 25 },
+  { id: 'uniqname', label: 'uniqname', minWidth: 100 },
+  { id: 'grade', label: 'grade', minWidth: 100 }
+]
+
+function ConfirmationTable (props: ConfirmationTableProps): JSX.Element {
+  // const classes = useStyles()
+  const [tableRows, setTableRows] = useState<StudentGrade[]>([])
+  const [page, setPage] = React.useState<number>(0)
+  const [rowsPerPage, setRowsPerPage] = React.useState(5)
+
+  useEffect(() => {
+    setTableRows(props.grades.map(i => {
+      return { rowNumber: i.rowNumber, uniqname: i.uniqname, grade: i.grade }
+    }).sort((a, b) => (a.rowNumber < b.rowNumber ? -1 : 1))
+    )
+  }, [props.grades])
+
+  const renderTable = (): JSX.Element => {
+    const emptyRows = rowsPerPage - Math.min(rowsPerPage, tableRows.length - page * rowsPerPage)
+
+    const handleChangePage = (event: unknown, newPage: number): void => {
+      setPage(newPage)
+    }
+
+    const handleChangeRowsPerPage = (event: { target: { value: string } }): void => {
+      setRowsPerPage(parseInt(event.target.value, rowsPerPage))
+      setPage(0)
+    }
```

Comment:
You can probably specify better event types here, like `React.ChangeEvent`. What you use depends on what prop you're passing to; see https://stackoverflow.com/questions/42081549/typescript-react-event-types

Reply:
Looks like you started doing that below.

Reply:
I  believe in this case the event was this b/c of the implementation of TablePagination (or some parent component?)

At least the code assistance identified it as "event: { target: { value: string } }" rather than some  specific React type event.  

I tried to use specific types were I could.  I'm not absolutely sure I didn't miss something, though.  I thought bringing in @types/material-ui might help but it didn't with this, at least.

Reply:
Huh, okay, bizarre. If you tried, you tried, gotta do something that works.